views/prisonerSearch/partials/prisonerSearchList.njk

```diff
@@ -1,7 +1,7 @@
 {% set rows = [] %}
 {% for prisoner in results %}
   {% set prisonerImage = '<img src="/app/images/' + prisoner.offenderNo + '/data" alt="Image of ' + prisoner.name + '" class="prisoner-search__results-list__image" />' %}
-  {% set prisonerLink = '<a href="/prisoner/' + prisoner.offenderNo + '" class="govuk-link">' + prisoner.name + '</a>'%}
+  {% set prisonerLink = '<a href="/save-backlink?service=digital-prison-services&returnResource=/prisoner-search%3Fkeywords%3D' + formValues.keywords + '%26location%3D' + formValues.location +  '%26alerts[]%3D' + formValues.alerts + '&toResource=/prisoner/' + prisoner.offenderNo + '" class="govuk-link">' + prisoner.name + '</a>'%}
```

Comment:
is this trying to capture the current url? Would it be safe to pass `req.url` through here instead?

Reply:
Looked into it and seems like since part of the URL is query params, it's gets confused unless the **returnResource**  has encoded values. Not sure if there's an out of the box query param encoder, or if we'd have to write a custom one. Think sticking with building up the url is best suited here?

Reply:
theres this kind of thing, and decoder
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Reply:
It's obviously quite fragile having to keep this url up to date with the application and bugs might be quite subtle a new filter is added but this isn't updated